rfcs/20240312-standardize-tan-customcall-tuplecollective-ops.md

```diff
@@ -0,0 +1,359 @@
+# [RFC] Standardize TanOp, CustomCallOp with typed FFI, and tuple-collectives ops
+
+Status: Draft<br/>
+Initial version: 03/12/2024<br/>
+Last updated: 03/12/2024<br/>
+Discussion thread: [GitHub](add PR Link)
+
+## Motivation
```

Comment:
Let's use the following skeleton, will group things better and make it easier to remove a section at a time if ever needed.

```
# RFC Motivation

# TanOp

## Rationale
// link existing ticket etc.

## Proposed Specification

### tan

#### Semantics

// ... the rest of tan spec

# CustomCallOp with typed FFI

## Rationale
// link existing ticket etc.

## Proposed Specification
// ....

# Tuple-collectives (AllGatherOp, AllReduceOp, AllToAllOp)

## Rationale

## Proposed spec

```